### ISSUE TITLE
Can not inherit from RoutlerOutlet (or from any directive ?)

### ISSUE THREAD

**Issue by fredgate** [angular/angular#12920]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I want to create a custom RouterOutlet.
For that I create a new directive that inherit from RouterOutlet and define a new selector.
```
@Directive({
	selector: 'custom-router-outlet',
	exportAs: 'customRouterOutlet'
})
export class CustomRouterOutlet extends RouterOutlet {
	constructor(
		parentOutletMap: RouterOutletMap,
		location: ViewContainerRef,
		resolver: ComponentFactoryResolver,
		@Attribute('name') name: string) {
		super(parentOutletMap, location, resolver, name);
		console.info("custom router outlet !!");
	}
}
```
Then I import this directive in my app module, and use it in the template of my app component :
```
 <custom-router-outlet></custom-router-outlet>
```
The application fail to load because an error occurs :

> Unhandled Promise rejection: Template parse errors:
> 'custom-router-outlet' is not a known element:
> 1. If 'custom-router-outlet' is an Angular component, then verify that it is part of this module.
> 2. If 'custom-router-outlet' is a Web Component then add "CUSTOM_ELEMENTS_SCHEMA" to the '@NgModule.schemas' of this component to suppress this message.

If I change the selector of my CustomRouterOutlet to set the same as RouterOutlet, and use <router-outlet> in my template, then the application load succesfully and my custom RouterOutlet is well used (message in the constructor appears in the console).
But if I add a template variable #outlet, then the application fails again with a message indicating that there is no directive with the exportAs _CustomRouterOulet_.
It seems that angular use the Directive annotation of the parent directive, and not that of the child directive.

I have the problem with the RouterOutlet directive but may be that the problem is more global and that it occurs with all inherited directives.

**Expected behavior**
The application should work and display the components in the custom router outlet.

**Minimal reproduction of the problem with instructions**
Here is a plunker showing the problem :
  http://plnkr.co/edit/mNJeZE2GAIaYWkkyn9cn?p=preview

**What is the motivation / use case for changing the behavior?**
Be able to use a custom router outlet.

**Please tell us about your environment:**
* **Angular version:**
2.1.1
**I had not the problem before when I used Angular 2 beta 17** (sorry I did not use intermediate versions like rc), so it seems a severe regression.

* **Browser:**
Tested with Firefox and Internet Explorer, but surely on all browsers.
 
* **Language:**
Tested with typescript.

* **Node (for AoT issues):** `node --version` =   
AoT not used.

**Comment by DzmitryShylovich** [comment]
See https://github.com/angular/angular/issues/11606

**Comment by fredgate** [comment]
I read the issue #11606 carrefully with attention, and I am not sure that it is the same thing.
In issue #11606 the problem is how are merged inputs available via decorators; according to the cases the inputs of the parent component can not be present in the child component. But the two components are usable.

My problem is diffrent and would be almost the reverse : the decorator of my child directive is never available and my child directive can be inaccessible : 
```
Unhandled Promise rejection: Template parse errors: **'custom-router-outlet' is not a known
element**:
If 'custom-router-outlet' is an Angular component, then verify that it is part of this module
If 'custom-router-outlet' is a Web Component then add "CUSTOM_ELEMENTS_SCHEMA" to
the '@NgModule.schemas' of this component to suppress this message.
```
On the other hand, my child directive is accessible using the selector of the parent directive. Worst : if I completely remove the @Directive decorator of my child directive, it is still accessible in the template... using the selector of the parent directive !

So my case is different : it is not about merging inherited decoartors. It is that **decorator of my child RouterOutlet is completely ignored** !
I do not know why, and I feel that the reason is different from the issue #11606. May be it is specific to router outlet.
Any ideas or explanations ?

**Comment by zoechi** [comment]
#11606 is an issue about all issues related to inheritance. If the component is not recognized because it inherits another component, then it will be addressed by #11606

**Comment by fredgate** [comment]
Ok. But It seems that the problem is different of this in #11606. So I will a comment in #11606 to expose the case.

**Comment by nhducseuit** [comment]
I know this is already closed by I'm facing this with 5.0.5 and reading #11606 doesn't help.

**Comment by nhducseuit** [comment]
Is there a solution or workaround for this?
